oximeter/db/src/client/mod.rs

```diff
@@ -737,21 +735,19 @@ impl Client {
 
         // Decide how to conform the on-disk version with this version of
         // Oximeter.
-        if version < expected_version {
-            info!(self.log, "wiping and re-initializing oximeter schema");
-            // If the on-storage version is less than the constant embedded into
-            // this binary, the DB is out-of-date. Drop it, and re-populate it
-            // later.
+        if version == 0 {
+            info!(self.log, "initializing oximeter schema");
+            // If the version is 0 the database does not exist.
+            // Create and populate it.
             if !replicated {
-                self.wipe_single_node_db().await?;
                 self.init_single_node_db().await?;
             } else {
-                self.wipe_replicated_db().await?;
                 self.init_replicated_db().await?;
             }
-        } else if version > expected_version {
+        } else if version != expected_version {
```

Comment:
No real action necessary, but this is part of why I think we probably want to move away from `oximeter` doing this work at all.

Suppose `oximeter` starts up, and is sitting at the loop waiting for the DB to be at its expected version. Meanwhile, we're separately upgrading the DB, _past_ `oximeter`'s expected version. This condition might feasibly be true only temporarily -- `oximeter` would then continue expecting the DB to be at its version, while the updater continues to move beyond it.

I'm not 100% sure what to do about this. We might need to commit to upgrading `oximeter` first, before applying any schema updates. We might also need some explicit way to release `oximeter`, so that it waits until the updater (the admin server, presumably) tells it to start operating.

Reply:
Sounds good, thanks @karencfv. Let me know if I can help or you need a rubber duck!

Reply:
Thanks! will do!